src/main/java/com/multithread/ForkJoinSum.java

```diff
@@ -0,0 +1,41 @@
+package com.multithread;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.ForkJoinTask;
+import java.util.concurrent.RecursiveTask;
+
+public class ForkJoinSum extends RecursiveTask<Long> {
+    private static final int THRESHOLD = 20;
+    private List<Integer> intList;
+
+    public ForkJoinSum(List<Integer> intList) {
+        this.intList = intList;
+    }
+
+    @Override
+    public Long compute() {
+        if (intList.size() > THRESHOLD) {
+            return ForkJoinTask.invokeAll(createSubtasks())
+                    .stream()
+                    .mapToLong(ForkJoinTask::join)
+                    .reduce(0, Long::sum);
+        } else {
+            return processing(intList);
```

Comment:
```suggestion
        } 
            return processing(intList);
```
we don't need else after return

Reply:
done